examples/atari/train_drqn_ale.py

```diff
@@ -0,0 +1,243 @@
+
+    def make_env(test):
+        # Use different random seeds for train and test envs
+        env_seed = test_seed if test else train_seed
+        env = atari_wrappers.wrap_deepmind(
+            atari_wrappers.make_atari(args.env, max_frames=args.max_frames),
+            episode_life=not test,
+            clip_rewards=not test,
+            flicker=args.flicker,
+            frame_stack=not args.no_frame_stack,
+        )
+        env.seed(int(env_seed))
+        if test:
+            # Randomize actions like epsilon-greedy in evaluation as well
+            env = chainerrl.wrappers.RandomizeAction(env, args.eval_epsilon)
+        if args.monitor:
+            env = gym.wrappers.Monitor(
+                env, args.outdir,
+                mode='evaluation' if test else 'training')
+        if args.render:
+            env = chainerrl.wrappers.Render(env)
+        return env
+
+    env = make_env(test=False)
+    eval_env = make_env(test=True)
+    print('Observation space', env.observation_space)
+    print('Action space', env.action_space)
+
+    n_actions = env.action_space.n
+    if args.recurrent:
+        # Q-network with LSTM
+        q_func = chainerrl.links.StatelessRecurrentSequential(
+            L.Convolution2D(None, 32, 8, stride=4),
+            F.relu,
+            L.Convolution2D(None, 64, 4, stride=2),
+            F.relu,
+            L.Convolution2D(None, 64, 3, stride=1),
+            functools.partial(F.reshape, shape=(-1, 3136)),
+            F.relu,
+            L.NStepLSTM(1, 3136, 512, 0),
+            L.Linear(None, n_actions),
+            DiscreteActionValue,
+        )
+        # Replay buffer that stores whole episodes
+        rbuf = replay_buffer.EpisodicReplayBuffer(10 ** 6)
+    else:
+        # Q-network without LSTM
+        q_func = chainer.Sequential(
+            L.Convolution2D(None, 32, 8, stride=4),
+            F.relu,
+            L.Convolution2D(None, 64, 4, stride=2),
+            F.relu,
+            L.Convolution2D(None, 64, 3, stride=1),
+            functools.partial(F.reshape, shape=(-1, 3136)),
+            L.Linear(None, 512),
+            F.relu,
+            L.Linear(None, n_actions),
+            DiscreteActionValue,
+        )
+        # Replay buffer that stores transitions separately
+        rbuf = replay_buffer.ReplayBuffer(10 ** 6)
+
+    # Draw the computational graph and save it in the output directory.
+    fake_obss = np.zeros(env.observation_space.shape, dtype=np.float32)[None]
+    if args.recurrent:
+        fake_out, _ = q_func(fake_obss, None)
+    else:
+        fake_out = q_func(fake_obss)
+    chainerrl.misc.draw_computational_graph(
+        [fake_out], os.path.join(args.outdir, 'model'))
+
+    explorer = explorers.LinearDecayEpsilonGreedy(
+        1.0, args.final_epsilon,
+        args.final_exploration_frames,
+        lambda: np.random.randint(n_actions))
+
+    opt = chainer.optimizers.Adam(1e-4, eps=1e-4)
```

Comment:
I'm assuming this should be fine. But why did you use Adam?

Reply:
It is just because Adam seems preferred in literature recently. I don't mean to reproduce any paper in this example.